device/device.go

```diff
@@ -162,8 +161,8 @@ func (d *Device) Connect(result chan<- error) {
 		policy.Reset()
 		connectOperation := func() error {
 			connectToken := d.m.Connect()
-			if ok := connectToken.WaitTimeout(30 * time.Second); !ok {
-				return errors.New("Timed out while connecting to the Broker.")
+			if connectToken.Wait() && connectToken.Error() != nil {
```

Comment:
I think you're ruling out the case in which connectToken.Wait() returns false and moving on